tests/framework/driver/data_driven_agent_test_driver.py

```diff
@@ -60,9 +61,75 @@ def one_test(self, hocon_file: str):
         """
         test_case: Dict[str, Any] = self.parse_hocon_test_case(hocon_file)
 
+        agent: str = test_case.get("agent")
+        self.asserts_basis.assertIsNotNone(agent)
+
+        # Get the success ration
+        success_ratio: str = test_case.get("success_ratio", "1/1")
+        self.asserts_basis.assertIn("/", success_ratio)
+
+        # Find the integer components of the success ratio
+        success_split: List[str] = success_ratio.split("/")
+        num_need_success: int = int(success_split[0])
+        num_iterations: int = int(success_split[-1])
+
+        # Put some bounds on the number of iterations
+        num_iterations = max(1, num_iterations)
+        num_need_success = min(num_need_success, num_iterations)
+
+        # Capture asserts for each iteration
+        iteration_asserts: List[AssertCapture] = []
+
+        # Loop through each iteration, capturing any asserts.
+        num_successful: int = 0
+        for index in range(num_iterations):
+
+            _ = index
+
+            # Capture the asserts for this iteration and add it to the list for later
+            assert_capture = AssertCapture(self.asserts_basis)
+            iteration_asserts.append(assert_capture)
+
+            # Perform a single iteration of the test.
+            self.one_iteration(test_case, assert_capture)
+
+            # Update our counter if this iteration is successful
+            asserts: List[AssertionError] = assert_capture.get_asserts()
+            if len(asserts) > 0:
+                # Not successful
+                continue
+
+            num_successful += 1
+            if num_successful == num_need_success:
+                # Don't do more tests than we actually need to
+                break
+
+        # Don't bother reporting any asserts if we have met our success ratio.
+        # Return early to pass this test.
+        if num_successful >= num_need_success:
+            return
+
+        # Find the first assert that fails and use it to fail this test
+        for assert_capture in iteration_asserts:
+            asserts: List[AssertionError] = assert_capture.get_asserts()
+            if len(asserts) > 0:
+                one_assert: AssertionError = asserts[0]
+                message: str = f"""
+{num_successful} of {num_iterations} iterations on agent {agent} were successful.
+Need at least {num_need_success} to consider {hocon_file} test to be successful.
+"""
+                raise AssertionError(message) from one_assert
+
+    def one_iteration(self, test_case: Dict[str, Any], asserts: AssertForwarder):
+        """
+        Perform a single iteration on the test case.
+
+        :param test_case: The dictionary describing the data-driven test case
+        :param asserts: The AssertForwarder to send asserts to.
+        """
```

Comment:
Note that the asserts are now passed in instead of being taken from the member variable.
That member variable now sticks around as the basis for the AssertCapture.